### ISSUE TITLE
[🐛] 🔥 A new message does not update preview list if app was in background once

### ISSUE THREAD

**Issue by RRaideRR** [GetStream/stream-chat-react-native#2583]
## Issue

The preview message is not updated when the user brings the app from the background to the foreground. It used to work e.g. with 5.26.0. See also the attached screencast. 

### Steps to reproduce
https://github.com/RRaideRR/getstream-preview-bug

### Expected behavior
The preview message is updated accordingly.

### Customization

<details><summary>Click To Expand</summary>
<p>

```typescript jsx
# N/A
```
</p>
</details>

### Offline support

- [X] I have enabled offline support.
- [ ] The feature I'm having does not occur when offline support is disabled. (stripe out if not applicable)

## Environment

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
{
  "name": "getstreambug",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@react-native-community/netinfo": "11.1.0",
    "@stream-io/flat-list-mvcp": "^0.10.3",
    "expo": "~50.0.14",
    "expo-file-system": "~16.0.8",
    "expo-image-manipulator": "~11.8.0",
    "expo-image-picker": "~14.7.1",
    "expo-media-library": "~15.9.1",
    "expo-status-bar": "~1.11.1",
    "react": "18.2.0",
    "react-native": "0.73.6",
    "react-native-gesture-handler": "~2.14.0",
    "react-native-quick-sqlite": "^8.0.6",
    "react-native-reanimated": "~3.6.2",
    "react-native-svg": "14.1.0",
    "stream-chat-expo": "^5.33.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0"
  },
  "private": true
}

```

**`react-native info` output:**

```
info Fetching system and libraries information...
(node:51861) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
System:
  OS: macOS 14.5
  CPU: (8) arm64 Apple M1
  Memory: 62.67 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.20.2
    path: ~/.nvm/versions/node/v18.20.2/bin/node
  Yarn:
    version: 1.22.19
    path: ~/.yarn/bin/yarn
  npm:
    version: 10.5.0
    path: ~/.nvm/versions/node/v18.20.2/bin/npm
  Watchman:
    version: 2024.04.29.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/opt/ruby/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK: Not Found
IDEs:
  Android Studio: 2023.3 AI-233.14808.21.2331.11709847
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.9
    path: /usr/bin/javac
  Ruby:
    version: 3.3.1
    path: /opt/homebrew/opt/ruby/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.6
    wanted: 0.73.6
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: true
  newArchEnabled: false

info React Native v0.74.3 is now available (your project is running on v0.73.6).
info Changelog: https://github.com/facebook/react-native/releases/tag/v0.74.3
info Diff: https://react-native-community.github.io/upgrade-helper/?from=0.74.3
info For more info, check out "https://reactnative.dev/docs/upgrading?os=macos".

```

- **Platform that you're experiencing the issue on**:
    - [ ] iOS
    - [ ] Android
    - [X] **iOS** but have not tested behavior on Android
    - [ ] **Android** but have not tested behavior on iOS
    - [ ] Both
- **`stream-chat-react-native` version you're using that has this issue:**
    - `stream-chat-expo: 5.33.0`
- Device/Emulator info:
  - [ ] I am using a physical device
  - OS version: `17.4`
  - Emulator: `iPhone 12 mini`

</p>
</details>

### Screenshots

<details><summary>Click To Expand</summary>
<p>

https://github.com/GetStream/stream-chat-react-native/assets/3411292/a036c57e-2b2d-485b-8df1-cdafa5be8a9c

</p>
</details>

---

**Comment by RRaideRR** [comment]
Additional debugging:

- `onNewMessage` at ChannelList is called
- last working version: 5.29.0 --> thus the bug was introduced with 5.30.0 

I'm a simple man: I see "channel preview" in the 5.30.0 release notes so maybe its related to the changes with #2509 ?

**Comment by khushal87** [comment]
Hey @RRaideRR, thanks for reporting the issue. I indeed introduced a regression, and this is fixed in #2585. Thanks again and, sorry for the inconvenience. :)
